presto-cache/src/test/java/com/facebook/presto/cache/alluxio/TestAlluxioCachingFileSystem.java

```diff
@@ -0,0 +1,411 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cache.alluxio;
+
+import alluxio.exception.ExceptionMessage;
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+import alluxio.util.io.FileUtils;
+import com.facebook.presto.cache.CacheConfig;
+import com.facebook.presto.cache.CacheFactory;
+import com.facebook.presto.hive.filesystem.ExtendedFileSystem;
+import io.airlift.units.DataSize;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PositionedReadable;
+import org.apache.hadoop.fs.Seekable;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.util.Progressable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.EOFException;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+
+import static com.facebook.presto.cache.CacheTestUtils.stressTest;
+import static com.facebook.presto.cache.CacheTestUtils.validateBuffer;
+import static com.facebook.presto.cache.CacheType.ALLUXIO;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static java.nio.file.Files.createTempDirectory;
+import static org.testng.Assert.assertEquals;
+
+@Test(singleThreaded = true)
+public class TestAlluxioCachingFileSystem
+{
+    private static final int DATA_LENGTH = (int) new DataSize(20, KILOBYTE).toBytes();
+    private static final int PAGE_SIZE = (int) new DataSize(1, KILOBYTE).toBytes();
+    private final byte[] data = new byte[DATA_LENGTH];
+    private URI cacheDirectory;
+    private String testFilePath;
+    private Map<String, Long> baseline = new HashMap<>();
+
+    @BeforeClass
+    public void setup()
+            throws IOException
+    {
+        new Random().nextBytes(data);
+        this.cacheDirectory = createTempDirectory("alluxio_cache").toUri();
+    }
+
+    @AfterClass
+    public void close()
+            throws IOException
+    {
+        checkState(cacheDirectory != null);
+        FileUtils.deletePathRecursively(cacheDirectory.getPath());
+    }
+
+    @BeforeMethod
+    public void setupMethod()
+    {
+        testFilePath = String.format("/test/file_%d", new Random().nextLong());
```

Comment:
use `createTempDirectory`?

Reply:
This path is only used for in memory stream and does not require creating a local file.

Reply:
Make a comment to add the "in memory stream" part to explain the path is never materialized

Reply:
added